### PR TITLE
[#3685] add tooltip translations

### PR DESCRIPTION
**Pull request by valllllll2000**
#### Reviewer Checklist
* [ ] Added an explanation about the work done
* [ ] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header (when relevant)
* [ ] Formatted the code
* [ ] Added a documentation (if relevant)
* [ ] Added some unit tests (if relevant)

### CODE REVIEW

**Review by muloem — COMMENTED** [review]
I think we could update the `FlowXmlObjectWriterTests` to take into account translations as well. And even add tests for the Form, question groups translations as well.

File: GAE/src/com/gallatinsystems/survey/dao/QuestionDao.java [hunk 1]
```diff
@@ -593,8 +593,8 @@ public Question getByKey(Long questionId) {
                     if (Question.Type.OPTION == q.getType()) {
                         q.setQuestionOptionMap(optionDao.listOptionByQuestion(q.getKey().getId()));
                     }
-                    q.setTranslationMap(translationDao.findTranslations(
-                            ParentType.QUESTION_TEXT, q.getKey().getId()));
+                    q.setTranslations(translationDao.findTranslations(
+                            q.getKey().getId(), ParentType.QUESTION_TEXT));
```
Comment (muloem) [inline GAE/src/com/gallatinsystems/survey/dao/QuestionDao.java:597]:
```suggestion
                            q.getKey().getId(), ParentType.QUESTION_TEXT, ParentType.QUESTION_TIP));
```

I believe in this one indeed we need to add the Question top as well.

**Comment by valllllll2000** [comment]
@muloem added the test

File: GAE/src/com/gallatinsystems/survey/domain/Question.java [hunk 2]
```diff
@@ -184,12 +185,26 @@ public void setQuestionGroupId(Long questionGroupId) {
         this.questionGroupId = questionGroupId;
     }
 
-    public Map<String, Translation> getTranslationMap() {
-        return translationMap;
+    public Map<String, Translation> translationsAsMap() {
```
Comment (valllllll2000) [inline GAE/src/com/gallatinsystems/survey/domain/Question.java:188]: this cannot be a get method or the map and list both appear in the json file

**Review by muloem — APPROVED** [review]

**Pull request merged**